sql-aggregates/profit-margin.sql

```diff
@@ -0,0 +1,21 @@
+with cte_inventory as (
+  select "filmId", "title", count("inventoryId") as "total_inventories"
+  from "inventory"
+  join "films" using ("filmId")
+  group by "filmId"
+),
+
+  cte_profit as (
+  select "title",
+          ("total_inventories" * "rentalDuration" * "rentalRate") as "profit",
+          ("total_inventories" * "replacementCost") as "cost"
+  from "cte_inventory"
+  join "films" using ("title")
+  group by "title", "total_inventories", "rentalRate", "rentalDuration", "replacementCost"
+)
+
+select "title", "profit", "cost"
+from "cte_profit"
+where "profit" > "cost"
+order by "profit" desc
+limit 5;
```

Comment:
Great job attempting this, but not quite the result we are looking for. You are definitely on the right track though using common table expressions! I would recommend making a cte that sums up each films total replacement costs, a second cte that calculates a films total revenue, then in your final query subtract the replacement costs from the total revenue of each film. I hope to see another attempt at this and lmk if you want any further hints!

Reply:
So what is the right answer? Can i have it to compare when i try to complete it again. Thanks!